mubench.pipeline/tasks/implementations/compile.py

```diff
@@ -158,25 +159,46 @@ def _compile(commands: List[str], project_dir: str, dep_dir: str) -> None:
         for command in commands:
             if command.startswith("mvn "):
                 command = "mvn dependency:build-classpath " + command[4:]
-                output = Shell.exec(command, cwd=project_dir,logger=logger)
-                dependencies = Compile._parse_maven_classpath(output)
+                output = Shell.exec(command, cwd=project_dir, logger=logger)
+                dependencies = Compile.__parse_maven_classpath(output)
                 Compile._copy_classpath(dependencies, dep_dir)
             elif command.startswith("ant"):
                 command += " -debug -verbose"
-                output = Shell.exec(command, cwd=project_dir,logger=logger)
-                dependencies = Compile._parse_ant_classpath(output)
+                output = Shell.exec(command, cwd=project_dir, logger=logger)
+                dependencies = Compile.__parse_ant_classpath(output)
                 Compile._copy_classpath(dependencies, dep_dir)
             elif command.startswith("gradle "):
-                Shell.exec(command,cwd=project_dir,logger=logger)
-                shutil.copy(os.path.dirname(__file__) + '/classpath.gradle', project_dir)
-                output = Shell.exec("gradle :printClasspath -b classpath.gradle", cwd=project_dir, logger=logger)
-                dependencies = Compile._parse_gradle_classpath(output)
+                Shell.exec(command, cwd=project_dir, logger=logger)
+                buildfile_dir = Compile.__parse_buildfile_dir(command)
+                shutil.copy(os.path.join(os.path.dirname(__file__), 'classpath.gradle'), os.path.join(project_dir, buildfile_dir))
+                command = "gradle :printClasspath -b '{}'".format(os.path.join(buildfile_dir, "classpath.gradle"))
+                output = Shell.exec(command, cwd=project_dir, logger=logger)
+                dependencies = Compile.__parse_gradle_classpath(output)
                 Compile._copy_classpath(dependencies, dep_dir)
             else:
                 Shell.exec(command, cwd=project_dir, logger=logger)
 
     @staticmethod
-    def _parse_maven_classpath(shell_output: str) -> List[str]:
+    def __parse_buildfile_dir(command):
+        # parsing for "-p" or "--project-dir" in gradle command
```

Comment:
I think this is clear from the `if "-p"/"--project-dir" in args` lines. Remove the comment.